core/src/main/java/org/elasticsearch/transport/netty/NettyTransportChannel.java

```diff
@@ -81,6 +87,9 @@ public void sendResponse(TransportResponse response) throws IOException {
 
     @Override
     public void sendResponse(TransportResponse response, TransportResponseOptions options) throws IOException {
+        if (reservation != null) {
+            reservation.doReturn();
```

Comment:
shouldn't this only be released in the `finally` clause of the following `try`? it seems like it'd be better only to release it after the response was sent